plugins/file/file_output.go

```diff
@@ -126,14 +151,28 @@ func (o *FileOutput) Init(config interface{}) (err error) {
 }
 
 func (o *FileOutput) openFile() (err error) {
-	basePath := filepath.Dir(o.Path)
+	basePath := filepath.Dir(o.path)
 	if err = os.MkdirAll(basePath, o.folderPerm); err != nil {
 		return fmt.Errorf("Can't create the basepath for the FileOutput plugin: %s", err.Error())
 	}
 	if err = plugins.CheckWritePermission(basePath); err != nil {
 		return
 	}
-	o.file, err = os.OpenFile(o.Path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, o.perm)
+	o.file, err = os.OpenFile(o.path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, o.perm)
+	return
+}
+
+func (o *FileOutput) tryRotateFile() (err error) {
+	if o.RotationInterval == 0 {
+		return
+	}
+	t := time.Now().Truncate(time.Duration(o.RotationInterval) * time.Hour)
```

Comment:
Yes, I already thought of you proposal, but it would introduce a second configuration parameter and as I was not sure about the performance penalty, I decided to not do premature optimization and keep the more simple solution for now.